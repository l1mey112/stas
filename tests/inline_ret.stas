include 'std.stas'

- inline
fn inline 1 0 {
	dup 10 > if {
		putu " ret\n" puts
		ret
	}
	dup putu " going on\n" puts
	dup 5 < if {
		putu " ret\n" puts
		ret
	}
	putu " end\n" puts
}

fn main {
	6  inline
	10 inline
	2  inline
	11 inline
	5  inline
	7  inline
}