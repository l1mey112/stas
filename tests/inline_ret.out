6 going on
6 end
10 going on
10 end
2 going on
2 ret
11 ret
5 going on
5 end
7 going on
7 end
